api/v1beta1/awsloadbalancercontroller_types.go

```diff
@@ -0,0 +1,206 @@
+
+const (
+
+	// AutoSubnetTaggingPolicy enables automatic subnet tagging.
+	AutoSubnetTaggingPolicy SubnetTaggingPolicy = "Auto"
+
+	// ManualSubnetTaggingPolicy disables automatic subnet tagging.
+	ManualSubnetTaggingPolicy SubnetTaggingPolicy = "Manual"
+)
+
+// AWSLoadBalancerControllerSpec defines the desired state of AWSLoadBalancerController
+type AWSLoadBalancerControllerSpec struct {
+
+	// SubnetTagging describes how resource tagging will be done by the operator.
+	//
+	// When in "Auto", the operator will detect the subnets where the load balancers
+	// will be provisioned and have the required resource tags on them. Whereas when
+	// set to manual, this responsibility lies on the user.
+	//
+	// +kubebuilder:default:=Auto
+	// +kubebuilder:validation:Optional
+	// +optional
+	SubnetTagging SubnetTaggingPolicy `json:"subnetTagging,omitempty"`
+
+	// Default AWS Tags that will be applied to all AWS resources managed by this
+	// controller (default []).
+	//
+	// This value is required so that this controller can function as expected
+	// in parallel to openshift-router.
+	//
+	// +kubebuilder:default:={}
+	// +kubebuilder:validation:Optional
+	// +optional
+	AdditionalResourceTags map[string]string `json:"additionalResourceTags,omitempty"`
+
+	// IngressClass specifies the Ingress class which the controller will reconcile.
+	// This Ingress class will be created unless it already exists.
+	// The value will default to "alb".
+	//
+	// +kubebuilder:default:=alb
+	// +kubebuilder:validation:Optional
+	// +optional
+	IngressClass string `json:"ingressClass,omitempty"`
+
+	// Config specifies further customization options for the controller's deployment spec.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	Config *AWSLoadBalancerDeploymentConfig `json:"config,omitempty"`
+
+	// AWSAddon describes the AWS services that can be integrated with
+	// the AWS Load Balancer.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	EnabledAddons []AWSAddon `json:"enabledAddons,omitempty"` // indicates which AWS addons should be disabled.
+
+	// Credentials is a reference to a secret containing
+	// the AWS credentials to be used by the controller.
+	// The secret is required to be in the operator namespace.
+	// If this field is empty - the credentials will be
+	// requested using the Cloud Credentials API,
+	// see https://docs.openshift.com/container-platform/4.11/authentication/managing_cloud_provider_credentials/about-cloud-credential-operator.html.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	Credentials *SecretReference `json:"credentials,omitempty"`
+}
+
+type AWSLoadBalancerDeploymentConfig struct {
+
+	// +kubebuilder:default:=2
+	// +kubebuilder:validation:Optional
+	// +optional
+	Replicas int32 `json:"replicas,omitempty"`
+}
+
+// SecretReference contains the information to let you locate the desired secret.
+// Secret is required to be in the operator namespace.
+type SecretReference struct {
```

Comment:
There was a similar (not entirely though) [discussion](https://github.com/openshift/aws-load-balancer-operator/pull/85#discussion_r1038628451) we had at the time I added the field. I guess my arguments didn't grow in number since then :)  I only have a future extensibility as a reasoning for a specific type. However since it's already a second time I'm suggested to use an existing type, I probably can go on an use `LocalSecretReference` or `SecretNameReference`. Especially if we'll do some other changes which would force me to add a conversion webhook.

Reply:
I'm wondering if the wording `Local` can be misleading. `AWSLoadBalancerController` CR is cluster scope, so it's not quite clear that `Local` refers to `LocalToOperator`. Maybe something like `SecretNameReference` would do the thing?

Reply:
Accepted, though unpredictable namespaces are a pain.